### PR TITLE
[MM-33640] storetest/team_store: use NewTestId for fields like names etc.

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
We are using `model.NewId()` for generating fields like `Name`, `DisplayName`, `Description` etc. When we are testing search functions there is a very small chance to have our search terms in the `Id`. To avoid that, we can use `NewTestId()` which manipulates the output of `model.NewId()` by replacing every odd character with a random digit.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-33640

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: store/storetest/team_store.go [hunk 1]
```diff
@@ -76,7 +76,7 @@ func TestTeamStore(t *testing.T, ss store.Store) {
 func testTeamStoreSave(t *testing.T, ss store.Store) {
 	o1 := model.Team{}
 	o1.DisplayName = "DisplayName"
-	o1.Name = "z-z-z" + model.NewId() + "b"
+	o1.Name = "z-z-z" + NewTestId() + "b"
```
Comment (agnivade) [inline store/storetest/team_store.go:79]: Can we also remove all of this "zz" business? ;)

**Review by agnivade — COMMENTED** [review]
Also, a test for the function would be good.

File: store/storetest/utils.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+// NewTestId is used for testing as a replacement for model.NewId(). It is a [A-Z0-9] string 26
+// characters long. It replaces every odd character with a digit.
+func NewTestId() string {
+	newId := []byte(model.NewId())
+
+	for i := 1; i < len(newId); i = i + 2 {
+		s := strconv.Itoa(rand.Intn(10))
+		newId[i] = s[0]
+	}
```
Comment (streamer45) [inline store/storetest/utils.go:26]: Do we have to randomize (and convert) further? We have everything we need in the original ID. What about:

```go
newId[i] = 48 + newId[i-1] % 10
```

This my inner [C](https://en.wikipedia.org/wiki/C_(programming_language)) heart suggesting :P

**Review by streamer45 — APPROVED** [review]
Nice :+1:

File: store/storetest/utils.go (hunk 2 above)
Comment (isacikgoz) [reply store/storetest/utils.go]: No resistance 👍

**Review by agnivade — APPROVED** [review]

**Pull request merged**